x-pack/plugin/sql/sql-client/src/test/java/org/elasticsearch/xpack/sql/client/HttpClientRequestTests.java

```diff
@@ -0,0 +1,313 @@
+        props.setProperty(ConnectionConfiguration.BINARY_COMMUNICATION, Boolean.toString(isBinary));
+        
+        URI uri = new URI(url);
+        ConnectionConfiguration conCfg = new ConnectionConfiguration(uri, url, props);
+        HttpClient httpClient = new HttpClient(conCfg);
+        
+        prepareMockResponse();
+        
+        try {
+            httpClient.basicQuery(query, fetchSize);
+        } catch (SQLException e) {
+            // we are not really interested if there is an exception
+        }
+        
+        assertEquals(1, webServer.requests().size());
+        RawRequest recordedRequest = webServer.takeRequest();
+        assertEquals(xContentType.mediaTypeWithoutParameters(), recordedRequest.getHeader("Content-Type"));
+        assertEquals("POST", recordedRequest.getMethod());
+        
+        BytesReference bytesRef = recordedRequest.getBodyAsBytes();
+        Map<String, Object> reqContent = XContentHelper.convertToMap(bytesRef, false, xContentType).v2();
+        
+        assertTrue(((String) reqContent.get("mode")).equalsIgnoreCase(Mode.CLI.toString()));
+        assertEquals(isBinary, reqContent.get("binary"));
+        assertEquals(Boolean.FALSE, reqContent.get("columnar"));
+        assertEquals(fetchSize, reqContent.get("fetch_size"));
+        assertEquals(query, reqContent.get("query"));
+        assertEquals("90000ms", reqContent.get("request_timeout"));
+        assertEquals("45000ms", reqContent.get("page_timeout"));
+        assertEquals("Z", reqContent.get("time_zone"));
+    }
+    
+    private void assertBinaryRequestForDrivers(boolean isBinary, XContentType xContentType) throws URISyntaxException {
+        String url = "http://" + webServer.getHostName() + ":" + webServer.getPort();
+        String query = randomAlphaOfLength(256);
+        Properties props = new Properties();
+        props.setProperty(ConnectionConfiguration.BINARY_COMMUNICATION, Boolean.toString(isBinary));
+        
+        URI uri = new URI(url);
+        ConnectionConfiguration conCfg = new ConnectionConfiguration(uri, url, props);
+        HttpClient httpClient = new HttpClient(conCfg);
+        
+        prepareMockResponse();
+        
+        Mode mode = randomFrom(Mode.JDBC, Mode.ODBC);
+        SqlQueryRequest request = new SqlQueryRequest(query, 
+                null,
+                ZoneId.of("Z"),
+                randomIntBetween(1, 100),
+                TimeValue.timeValueMillis(randomNonNegativeLong()),
+                TimeValue.timeValueMillis(randomNonNegativeLong()),
+                null,
+                randomBoolean(),
+                randomAlphaOfLength(128),
+                new RequestInfo(mode),
+                randomBoolean(),
+                randomBoolean(),
+                isBinary);
+        try {
+            httpClient.query(request);
+        } catch (SQLException e) {
+            // we are not really interested if there is an exception
```

Comment:
I would say to maybe move the logger to upper level and use it here, just in case.